transfer_client_portfolio/wizard/transfer_portfolio_wizard.py

```diff
@@ -0,0 +1,296 @@
+            res.update(
+                {
+                    "is_server_action": True,
+                    "server_action_type": "lead",
+                    "review_state": True,
+                    "opportunity_ids": [(6, 0, opportunity_ids)],
+                    "contact_ids": [
+                        (
+                            6,
+                            0,
+                            contact_ids,
+                        )
+                    ],
+                    "activity_ids": [(6, 0, activity_ids)],
+                }
+            )
+        elif self.env.context.get("is_partner_server_action"):
+            partner_ids = self.env.context.get("active_ids")
+            opportunity_ids = (
+                self.env["res.partner"]
+                .browse(partner_ids)
+                .mapped("opportunity_ids")
+                .ids
+            )
+            activity_ids = (
+                self.env["crm.lead"].browse(opportunity_ids).activity_ids
+                | self.env["res.partner"].browse(partner_ids).activity_ids
+            ).ids
+            res.update(
+                {
+                    "is_server_action": True,
+                    "server_action_type": "partner",
+                    "review_state": True,
+                    "contact_ids": [(6, 0, partner_ids)],
+                    "opportunity_ids": [(6, 0, opportunity_ids)],
+                    "activity_ids": [(6, 0, activity_ids)],
+                }
+            )
+        return res
+
+    def review_transfer(self):
+        self.ensure_one()
+        self.contact_ids = self.env["res.partner"].search(
+            [("user_id", "=", self.current_salesperson.id)]
+        )
+        self.opportunity_ids = self.env["crm.lead"].search(
+            [
+                ("user_id", "=", self.current_salesperson.id),
+                ("stage_id.allow_transfer_opportunity", "=", True),
+            ]
+        )
+        self.activity_ids = (
+            self.opportunity_ids.activity_ids | self.contact_ids.activity_ids
+        )
+        self.review_state = True
+        return {
+            "name": "Transfer Portfolio",
+            "view_mode": "form",
+            "view_id": False,
+            "res_model": self._name,
+            "domain": [],
+            "context": dict(self._context, active_ids=self.ids),
+            "type": "ir.actions.act_window",
+            "target": "new",
+            "res_id": self.id,
+        }
+
+    def transfer_portfolio(self):
+        if not self.new_salesperson:
+            raise ValidationError(_("You must select a new salesperson."))
+        transfer_ids = self.filtered(lambda x: x.contact_ids or x.opportunity_ids)
+        if not transfer_ids:
+            raise ValidationError(_("There are no records to transfer."))
+        if self.update_salesperson_contact:
+            contact_child_ids = self.env["res.partner"].search(
+                [("parent_id", "in", self.contact_ids.ids)]
```

Comment:
Why this search does not use commercial_partner_id instead of parent_id?

Reply:
I don't know because all in the first commit is very old, this is a change of repository. ¿Can we ignore it?

Reply:
No problem